### DISCUSSION TITLE
Is position duration tracking supported?

### DISCUSSION THREAD

**Discussion by michaelzus** [ib-api-reloaded/ib_async#37] [Q&A] [answered]
Hi all,

is there a way to track for how long the position is open ?
or get the open date/timestamp ?

Thank you.

**Comment by gnzsnz** [comment, marked as answer]
I would suggest to try `IB.fills()`, it will tell you open/close times. It has a "but", you need to aggregate and match fills

**Reply by michaelzus** [reply]
great, I'll try it.
Alternatively I'll just use some local db to log it, even something simple as csv.

**Comment by mattsta** [comment]
Also worth noting the IBKR API only returns executions and fills from the past 1-2 days at most before deleting them forever from the API output.

So your choices are:

- record every execution from the API (live or at the end of every day)
- use the IBKR web portal for pulling longer history reports
- parse the daily trade report emails

Also, if you are running multiple client ids for your API clients, only the client id where a trade executes will receive a live execution notification, but you can fetch all executions/fills for all clients at any time using `reqExecutionsAsync()` again.
